ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts

```diff
@@ -121,43 +123,48 @@ export class ShippingSearchComponent implements OnInit {
       ); // need to subscribe, otherwise it will not send!
   }
 
-  valueChanged( value: any, parameterName: string, kitRequest: KitRequest ): void {
-    let v;
-
+  //Triggers when user finishes typing
+  valueChanged(value: any, parameterName: string, kitRequest: KitRequest): void {
     if (typeof value === 'string') {
-      kitRequest[ parameterName ] = value;
-      v = value;
+      kitRequest[parameterName] = value;
     }
-    if (v != null) {
+  }
+
+  patch(patch: any, kitRequest: KitRequest): Observable<any> {
+    return this.dsmService.patchParticipantRecord(JSON.stringify(patch));
+  }
+
+  //Fires when user clicks "Save Date" button. Returns true
+  //if the promise succeeds and the server successfukky writes and
+  //returns and error if the server write fails.
+  saveDate(kitRequest: KitRequest): any {
```

Comment:
I suppose the return type here should be `Promise<any>`

Reply:
My apologies, I should have been more specific. Have a look at this code, that should give you a hint:
`saveDate(kitRequest: KitRequest): () => Promise<boolean> {
    return async (): Promise<boolean> => {... }}`